### PR TITLE
TASK: Remove base prototype and template path from autogenerated fusion for Node and Document

### PR DESCRIPTION
**Pull request by mficzel**
The base prototypes and templatePathes are not set by the prototypeGenrator for Neos.Neos:Node and Neos.Neos:Document any more.  For convenience reasons the prototypes for each NodeTypes are declared without base prototype and template all node properties are made accessible as fusion props.

### CODE REVIEW

**Comment by mficzel** [comment]
@gerhard-boden this relates to your pr  https://github.com/neos/neos-development-collection/pull/615

**Comment by gerhard-boden** [comment]
Ah nice, looks like a decision to go ahead and remove it has come faster than we both hopped 😄 

sooo, we can kick this as well? https://github.com/neos/neos-development-collection/blob/master/Neos.Neos/Classes/Domain/Service/DefaultDocumentPrototypeGenerator.php

**Comment by mficzel** [comment]
I was reluctant to remove the class because of the @api in there

**Comment by gerhard-boden** [comment]
hmmm... deprecate it as API? I wonder if anybody really uses that, and after deprecation the loss can be easily reconstituted since it's practically a one liner. But seems like dead weight to leave it in the code base like this forever. 

Or do we have a rule to never ever deprecate any classes with @api annotation?

**Comment by mficzel** [comment]
Add the @deprecate notice

**Comment by mficzel** [comment]
@gerhard-boden i adjusted this pr quite a bit so please review again. Instead of disabeling the autocreation now the protoype and the propertie-acessors are created bur template path and base prototype are removed vor Node and Document.

File: Neos.Neos/Classes/Domain/Service/DefaultPrototypeGenerator.php [hunk 1]
```diff
@@ -44,11 +51,17 @@ public function generate(NodeType $nodeType)
             return '';
         }
 
-        $output = 'prototype(' . $nodeType->getName() . ') < prototype(' . $this->basePrototypeName . ') {' . chr(10);
+        $output = 'prototype(' . $nodeType->getName() . ')';
+        if ($this->basePrototypeName) {
```
Comment (hlubek) [inline Neos.Neos/Classes/Domain/Service/DefaultPrototypeGenerator.php]: I'd make this more explicit with `$this->basePrototypeName !== null`

File: Neos.Neos/Classes/Domain/Service/DefaultPrototypeGenerator.php [hunk 2]
```diff
@@ -44,11 +51,17 @@ public function generate(NodeType $nodeType)
...
+            $output .= ' < prototype(' . $this->basePrototypeName . ')';
+        }
+        $output .= ' {' . chr(10);
 
-        list($packageKey, $relativeName) = explode(':', $nodeType->getName(), 2);
-        $templatePath = 'resource://' . $packageKey . '/Private/Templates/NodeTypes/' . $relativeName . '.html';
-        $output .= "\t" . 'templatePath = \'' . $templatePath . '\'' . chr(10);
+        if ($this->templatePath) {
```
Comment (hlubek) [inline Neos.Neos/Classes/Domain/Service/DefaultPrototypeGenerator.php]: Same here with `$this->templatePath !== null`

**Comment by mficzel** [comment]
@hlubek thanks, i adjusted the checks as requested

**Comment by gerhard-boden** [comment]
Thanks for answering during the weekly

**Pull request merged**